packages/react-core/src/components/Button/Button.tsx

```diff
@@ -130,12 +150,15 @@ export const Button: React.FunctionComponent<ButtonProps> = ({
       type={isButtonElement ? type : null}
       {...ouiaProps}
     >
-      {variant !== ButtonVariant.plain && icon && iconPosition === 'left' && (
-        <span className={css(styles.buttonIcon, styles.modifiers.start)}>{icon}</span>
-      )}
-      {children}
-      {variant !== ButtonVariant.plain && icon && iconPosition === 'right' && (
-        <span className={css(styles.buttonIcon, styles.modifiers.end)}>{icon}</span>
+      {isLoading ? (
```

Comment:
i'd suggest leaving the previous code and just adding this piece in the beginning, seems more readable that way to me
```
{isLoading && (
  <span className={css(styles.buttonProgress)}>
    <Spinner size={spinnerSize.md} aria-valuetext={spinnerAriaValueText} />
  </span>
)
```